README.md

```diff
@@ -202,44 +255,111 @@ the `lookup` result will sometimes provide a new array, and sometimes not.
   supported.
 * **label** [String][]: Optional label in which to store the result.
 
-## `execute()`
+**Returns:** `this<SetupChain>`
 
-Executes each task added sequentially and collects the results into a single object.
-The results for `execute` are returned. The internal state will be Persisted.
+```javascript
+const SetupChain = require('@logdna/setup-chain')
 
-**returns**: [Object][]
+class MyChain extends SetupChain {
+  constructor(state) {
+    super(state, {
+      hello: async () => {
+        return 'hi'
+      }
+    })
+  }
+}
 
-```javascript
-await new SetupChain()
-  .foo()
-  .bar()
-  .insertDBRecord({}, 'record')
+const chain = new MyChain()
+chain
+  .repeat(5, 'hello', {}, 'result')
   .execute()
+  .then((result) => {
+    console.log(result)
+  })
 
-{
-  foo: 'bar'
-, bar: 'foo'
-, record: { id: 1 }
-}
+> {result: ['hi', 'hi', 'hi', 'hi', 'hi']}
 ```
 
-## `sleep(<Object>)`
+## `sleep(opts)`
+
+When used in a chain, it will wait for a specified number of milliseconds before returning. Since this is a chain action, the sleep will appear to happen sequentially
+within the action chain.
 
-### Parameters
+**Parameters**
 
 * **opts** [Object][]: configuration options
   * **ms** [Number][]: The number of milliseconds to wait before continuing
 
-This is a builder function for the `sleep` action.
-It will Waits for a specified number of milliseconds before returning.
+**Returns**: `this<SetupChain>`
 
-**returns**: SetupChain
+```javascript
+const chain = new SetupChain()
+const state = await chain
+  .set('start', '!date:getTime')
+  .sleep({ms: 1000})
+  .set('delayed_1', '!date:getTime')
+  .sleep({ms: 1000})
+  .set('delayed_2', '!date:getTime')
+  .execute()
+
+> {start: 1620496433509, delayed_1: 1620496434512, delayed_2: 1620496435514, set: undefined, sleep: undefined}
+```
 
 ## `reset()`
 
 Manually clears internal state and any pending tasks
 
-**returns**: SetupChain
+**Returns**: `this<SetupChain>`
+
+## `execute()`
+
+Resolves the promises returned by chain actions, and returns the result
+of each action in a single object. This should be the final
+method in any chain, otherwise the actions will not be resolved into values.
+After `execute` is called, the return value is also stored in `chain.state` for
+access later, even if the return value is not immediately used.
+
+**Returns**: [Object][]
+
+## `lookup(input)`
+
+Resolves lookup value. This function is generally used by actions to resolve values
```

Comment:
I think the first sentence is not necessary if you only have to go on to explain what it means
```suggestion
This function is generally used by actions to resolve values
```